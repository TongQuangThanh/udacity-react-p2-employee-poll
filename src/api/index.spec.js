import { render, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import AddPool from "../Pages/Pools/AddPool";
import { _saveQuestion, _saveQuestionAnswer, formatQuestion } from "./index";
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);
const optionOneText = "have horrible short term memory";
const optionTwoText = "have horrible long term memory";
const author = "sarahedo";
const authenticatedUser = author;
const qid = "8xf0y6ziyjabvozdd253nd";
const answer = "optionOne";
const dumQuestion = { optionOneText, optionTwoText, author };

const store = mockStore({
  message: 'sample text',
});

describe('Snapshot testing', () => {
  it('will return correct Add pool page', () => {
    const component = render(
      <Provider store={store}>
        <AddPool />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});

describe('DOM test which uses the fireEvent function', () => {
  it('will return correct Add pool page', () => {
    const component = render(
      <Provider store={store}>
        <AddPool />
      </Provider>
    );
    const submitButton = component.getByTestId('submit-button');
    fireEvent.click(submitButton);
    expect(component.getByTestId('invalid-feedback')).toBeInTheDocument();
  });
});

describe('formatQuestion', () => {
  it('will return Question in format', () => {
    const results = formatQuestion(dumQuestion);
    expect(results.optionOne.votes.length).toEqual(0);
    expect(results.optionTwo.votes.length).toEqual(0);
    expect(results.optionOne.text).toEqual(optionOneText);
    expect(results.optionTwo.text).toEqual(optionTwoText);
  });
});

describe('_saveQuestion', () => {
  it('will return correct question format', async () => {
    const results = await _saveQuestion(dumQuestion);
    expect(results.optionOne.votes.length).toEqual(0);
    expect(results.optionTwo.votes.length).toEqual(0);
    expect(results.optionOne.text).toEqual(optionOneText);
    expect(results.optionTwo.text).toEqual(optionTwoText);
  });

  it('will return error when it not has author info', async () => {
    _saveQuestion({
      optionOneText,
      optionTwoText,
    }).catch(
      err => expect(err).toEqual("Question info not complete")
    );
  });
});


describe('_saveQuestionAnswer', () => {
  it('will return correct question format', async () => {
    const results = await _saveQuestionAnswer({ authenticatedUser, qid, answer });
    expect(results.optionOne.votes.length).toEqual(2);
    expect(results.optionTwo.votes.length).toEqual(0);
    expect(results.optionOne.text).toEqual(optionOneText);
    expect(results.optionTwo.text).toEqual(optionTwoText);
  });

  it('will return error when it not has Answer info', async () => {
    _saveQuestionAnswer({ authenticatedUser, qid, answer: null })
      .catch(err => expect(err).toEqual("Answer info not complete"));
  });
});
