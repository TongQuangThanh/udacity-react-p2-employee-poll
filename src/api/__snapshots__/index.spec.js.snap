// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing will return correct Add pool page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container min-h-100 mb-3"
      >
        <div
          id="form_container"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-lg-4"
            >
              <div
                id="left_form"
              >
                <figure>
                  <img
                    alt=""
                    height="100"
                    src="https://static.vecteezy.com/system/resources/previews/022/100/698/non_2x/marvel-logo-transparent-free-png.png"
                    style="background-color: rgb(255, 255, 255);"
                    width="100"
                  />
                  <p>
                    Creation of a new Pool
                    <br />
                    Please Enter The Options
                  </p>
                </figure>
              </div>
            </div>
            <div
              class="col-lg-8"
            >
              <div
                class="wizard"
                id="wizard_container"
              >
                <form
                  class="fl-form fl-style-1 wizard-form"
                  id="wrapped"
                >
                  <div
                    class="wizard-branch wizard-wrapper"
                    id="middle-wizard"
                  >
                    <div
                      class="step wizard-step current"
                    >
                      <h3
                        class="main_question"
                      >
                        <i
                          class="arrow_right"
                        />
                        Would You Rather ?
                      </h3>
                      <div
                        class="form-group"
                      >
                        <label
                          class=""
                          for="optionOne"
                        >
                          Option One
                        </label>
                        <input
                          aria-invalid="false"
                          class="form-control"
                          id="optionOne"
                          name="optionOne"
                          type="text"
                          value=""
                        />
                        <small
                          class="form-text text-muted"
                        >
                          Please Provide The First Option
                        </small>
                      </div>
                      <div
                        class="form-group"
                      >
                        <label
                          class=""
                          for="option2"
                        >
                          Option Two
                        </label>
                        <input
                          aria-invalid="false"
                          class="form-control"
                          id="optionTwo"
                          name="optionTwo"
                          type="text"
                          value=""
                        />
                        <small
                          class="form-text text-muted"
                        >
                          Please Provide The Second Option
                        </small>
                      </div>
                    </div>
                  </div>
                  <div
                    id="bottom-wizard"
                  >
                    <button
                      class="forward"
                      data-testid="submit-button"
                      name="forward"
                      type="submit"
                    >
                      Submit
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container min-h-100 mb-3"
    >
      <div
        id="form_container"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-lg-4"
          >
            <div
              id="left_form"
            >
              <figure>
                <img
                  alt=""
                  height="100"
                  src="https://static.vecteezy.com/system/resources/previews/022/100/698/non_2x/marvel-logo-transparent-free-png.png"
                  style="background-color: rgb(255, 255, 255);"
                  width="100"
                />
                <p>
                  Creation of a new Pool
                  <br />
                  Please Enter The Options
                </p>
              </figure>
            </div>
          </div>
          <div
            class="col-lg-8"
          >
            <div
              class="wizard"
              id="wizard_container"
            >
              <form
                class="fl-form fl-style-1 wizard-form"
                id="wrapped"
              >
                <div
                  class="wizard-branch wizard-wrapper"
                  id="middle-wizard"
                >
                  <div
                    class="step wizard-step current"
                  >
                    <h3
                      class="main_question"
                    >
                      <i
                        class="arrow_right"
                      />
                      Would You Rather ?
                    </h3>
                    <div
                      class="form-group"
                    >
                      <label
                        class=""
                        for="optionOne"
                      >
                        Option One
                      </label>
                      <input
                        aria-invalid="false"
                        class="form-control"
                        id="optionOne"
                        name="optionOne"
                        type="text"
                        value=""
                      />
                      <small
                        class="form-text text-muted"
                      >
                        Please Provide The First Option
                      </small>
                    </div>
                    <div
                      class="form-group"
                    >
                      <label
                        class=""
                        for="option2"
                      >
                        Option Two
                      </label>
                      <input
                        aria-invalid="false"
                        class="form-control"
                        id="optionTwo"
                        name="optionTwo"
                        type="text"
                        value=""
                      />
                      <small
                        class="form-text text-muted"
                      >
                        Please Provide The Second Option
                      </small>
                    </div>
                  </div>
                </div>
                <div
                  id="bottom-wizard"
                >
                  <button
                    class="forward"
                    data-testid="submit-button"
                    name="forward"
                    type="submit"
                  >
                    Submit
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
